File: LeetCode/p198.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int pre1 = 0, pre2 = 0;
        for (int x : nums) {
            int now = max(pre1, pre2 + x);
            pre2 = pre1;
            pre1 = now;
        }
        return pre1;
    }
};

File: LeetCode/p198.go
func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

func rob(nums []int) int {
    last1, last2 := 0, 0
    for _, v := range nums {
        t := max(last2 + v, last1)
        last1, last2 = t, last1
    }
    return max(last1, last2)
}
